src/components/Questions/QuestionModal.vue

```diff
@@ -305,10 +313,10 @@ export default defineComponent({
         return false
       }
       if (isQuestionTypeSubjective.value) return currentDraftResponse != ""
-      if (typeof currentDraftResponse == "number") {
+      if (isQuestionTypeNumericalInteger.value || isQuestionTypeNumericalFloat.value) {
         return true
       }
-      return currentDraftResponse.length > 0
+      return Object.keys(currentDraftResponse).length > 0
```

Comment:
Didnt understand why this was added

Reply:
because `.length` was giving a type error. `curentDraftResponse` need not always be an array or string.

Reply:
I just feel like `Object.keys()` is a bit misleading, even if it works in this case.
Basically we want to check for length because we're assuming that `currentDraftResponse` is an array.
So we should check for that as well.
`Object.keys("abcd").length` also gives a value of 4. I know that when the compiler reaches here, we can be sure that the question is not of subjective type but this line of code is a little misleading.
You can write this 
`Array.isArray(currentDraftResponse) && currentDraftResponse.length > 0`
I tried and it's not giving any error and the code is foolproof. It will always check the length if it's an array.